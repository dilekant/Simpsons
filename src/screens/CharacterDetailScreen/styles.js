import {StyleSheet} from 'react-native';
import {responsiveScreenHeight} from 'react-native-responsive-dimensions';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    height: responsiveScreenHeight(35),
    marginTop: 25,
    marginBottom: 15,
  },
  nameText: {
    color: '#000000',
    fontSize: responsiveScreenHeight(3.5),
    textAlign: 'center',
  },
  jobText: {
    fontSize: responsiveScreenHeight(2.2),
    textAlign: 'center',
  },
  aboutText: {
    marginTop: 15,
    fontSize: responsiveScreenHeight(1.8),
    marginHorizontal: 15,
  },
});

export default styles;
