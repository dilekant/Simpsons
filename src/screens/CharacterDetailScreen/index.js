import React, {useEffect, useState} from 'react';
import {View, Text, Image, ScrollView} from 'react-native';
import styles from './styles';

const CharacterDetailScreen = ({navigation, route}) => {
  const [detail, setDetail] = useState([]);

  useEffect(() => {
    setDetail(route.params?.detail);
  }, [route]);

  return (
    <ScrollView style={styles.container}>
      <View style={styles.container}>
        <Image
          resizeMode="contain"
          source={{uri: detail?.avatar}}
          style={styles.image}
        />
        <Text style={styles.nameText}>{detail?.name}</Text>
        <Text style={styles.jobText}>{detail?.job}</Text>
        <Text style={styles.aboutText}>{detail?.about}</Text>
      </View>
    </ScrollView>
  );
};

export default CharacterDetailScreen;
