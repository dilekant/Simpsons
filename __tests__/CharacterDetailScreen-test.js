import React from 'react';
import renderer from 'react-test-renderer';
import CharacterDetailScreen from '../src/screens/CharacterDetailScreen';

test('renders correctly', () => {
  const tree = renderer.create(<CharacterDetailScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});