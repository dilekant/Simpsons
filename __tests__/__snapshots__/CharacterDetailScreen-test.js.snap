// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": 466.9,
            "marginBottom": 15,
            "marginTop": 25,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 46.690000000000005,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 29.348000000000003,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "gray",
            "fontSize": 24.012000000000004,
            "marginHorizontal": 15,
            "marginTop": 15,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
